File: C++/119. Pascal's Triangle II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> sol = {{1},{1,1}};
        for(int i = 2; i<=rowIndex;i++){
            vector<int> temp;
            temp.push_back(1);
            for(int j = 0; j<sol[sol.size()-1].size()-1;j++){
                temp.push_back(sol[i-1][j]+sol[i-1][j+1]);
            }
            temp.push_back(1);
            sol.push_back(temp);
        }
        return sol[rowIndex];
    }
};

File: Java/119. Pascal's Triangle II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> arr = new ArrayList<>();
        for (int i = 0; i <= rowIndex; i++) {
            arr.add(0,1);
            for(int j = 1; j < i; j ++) {
                arr.set(j, arr.get(j) + arr.get(j + 1));
            }
        }
        return arr;
    }
}
